nixos/modules/services/web-apps/documenso.nix

```diff
@@ -0,0 +1,204 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+
+let
+  cfg = config.services.documenso;
+in
+{
+  options.services.documenso = {
+
+    enable = lib.mkEnableOption "Documenso service";
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.documenso;
+      defaultText = lib.literalExpression "pkgs.documenso";
+      description = ''
+        The documenso package to use.
+      '';
+    };
+
+    serverPort = lib.mkOption {
+      type = lib.types.port;
+      default = 3000;
+      description = "Port the documenso server listens on.";
+    };
+
+    database = {
+      name = lib.mkOption {
+        type = lib.types.str;
+        default = "documenso";
+        description = "Database name.";
+      };
+
+      user = lib.mkOption {
+        type = lib.types.str;
+        default = "documenso";
+        description = "Database user.";
+      };
+
+      url = lib.mkOption {
+        type = lib.types.str;
+        default = "postgres://${config.services.documenso.database.user}@localhost/${config.services.documenso.database.name}?host=/run/postgresql";
+        defaultText = lib.literalExpression ''
+          "postgres://documenso@localhost/documenso?host=/run/postgresql";
+        '';
+        description = ''
+          Database url. Note that any secret here would be world-readable. Use
+          `services.documenso.environmentFile` instead to include database
+          secrets.
+        '';
+      };
+
+      createLocally = lib.mkOption {
+        type = lib.types.bool;
+        default = true;
+        description = "Whether to create a local database automatically.";
+      };
+
+      seedDatabase = lib.mkOption {
+        type = lib.types.bool;
+        default = true;
+        description = ''
+          Whether to seed the database with initial users and teams and
+          templates. If you set this to 'true' you can login after first
+          install with `admin@documenso.com` and `password` as password.
+        '';
+      };
+
+    };
+
+    authSecret = lib.mkOption {
+      type = lib.types.str;
+      default = "secret";
```

Comment:
These should be paths right ?
Otherwise the secrets leak into the nix store which is public .